lib/elixir/lib/code/formatter.ex

```diff
@@ -2449,4 +2472,43 @@ defmodule Code.Formatter do
   defp has_double_quote?(chunk) do
     is_binary(chunk) and chunk =~ @double_quote
   end
+
+  # Migration rewrites
+
+  @bool_operators [
+    :>,
+    :>=,
+    :<,
+    :<=,
+    :in
+  ]
+  @guards [
+    :is_atom,
+    :is_boolean,
+    :is_nil,
+    :is_number,
+    :is_integer,
+    :is_float,
+    :is_binary,
+    :is_list,
+    :is_map,
+    :is_struct,
+    :is_function,
+    :is_reference,
+    :is_pid
+  ]
+
+  defp negate_condition(condition) do
+    case condition do
+      {neg, _, [condition]} when neg in [:!, :not] -> condition
+      {:|>, _, _} -> {:|>, [], [condition, {{:., [], [Kernel, :!]}, [closing: []], []}]}
+      {op, _, [_, _]} when op in @bool_operators -> {:not, [], [condition]}
+      {guard, _, [_ | _]} when guard in @guards -> {:not, [], [condition]}
```

Comment:
a thought i had while implementing this for styler -- being smart and using `not` here might be dangerous, because it's possible the user did something whacky like
```
import Kernel, except: [is_atom: 1]

def is_atom(...), do: :ok
```
better to just always use `!` and leave it to the user to translate to `not` as they want?

Reply:
The idea behind migrate option is that they rewrite the AST and cannot guarantee to work in edge cases where metaprogramming assumes a given AST or Kernel overwrites are done (after all people can have re-defined `unless` itself, or `==`, etc...).

This particular one feels like an unlikely case, the need to overwriting `is_atom` feels low, the convention that only guards returning booleans should start with `is_` should also discourage it... I think we're fine.